XLS-0056d-batch/README.md

```diff
@@ -0,0 +1,626 @@
+* Withdrawing accounts (multi-account): Attempt a withdrawal from your checking account, and if that fails, withdraw from your savings account instead.
+
+## 1. Overview
+
+This spec proposes one new transaction: `Batch`. It also proposes an addition to the common fields of all transactions. It will not require any new ledger objects, nor modifications to existing ledger objects. It will require an amendment, tentatively titled `featureBatch`.
+
+The rough idea of this design is that users can include "sub-transactions" inside `Batch`, and these transactions are processed atomically. The design also supports transactions from different accounts in the same `Batch` wrapper transaction.
+
+### 1.1. Terminology
+* **Inner transaction**: the sub-transactions included in the `Batch` transaction, that are executed atomically.
+* **Outer transaction**: the wrapper `Batch` transaction itself.
+* **Batch mode** or **mode**: the "mode" of batch processing that the transaction uses. See section 2.2 for more details.
+
+## 2. Transaction: `Batch`
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`TransactionType`|✔️|`string`|`UInt16`|
+|`Account`|✔️|`string`|`STAccount`|
+|`Fee`|✔️|`string`|`STAmount`|
+|`Flags`|✔️|`number`|`UInt32`|
+|`RawTransactions`|!|`array`|`STArray`|
+|`TxnIDs`|✔️|`array`|`Vector256`|
+|`BatchSigners`| |`array`|`STArray`|
+
+<!--
+```typescript
+{
+    TransactionType: "Batch",
+    Account: "r.....",
+    Flags: "1",
+    TxnIDs: [transaction hashes...]
+    RawTransactions: [transaction blobs...], // not included in the signature or stored on ledger
+    BatchSigners: [ // only sign the list of transaction hashes and probably the batch mode
+      BatchSigner: {
+        Account: "r.....",
+        Signature: "...."
+      },
+      BatchSigner: {
+        Account: "r.....",
+        Signers: [...] // multisign
+      },
+      ...
+    ],
+    SigningPubKey: "....",
+    TxnSignature: "...."
+}
+```
+-->
+
+### 2.1. `Fee`
+
+The fee for the outer transaction is:
+
+$$(n+2)*base\textunderscore fee + \sum_{innerTxns} Txn.Fee$$
+(where `n` is the number of signatures included in the outer transaction)
+
+In other words, the fee is twice the base fee (a total of 20 drops when there is no fee escalation), plus the sum of the transaction fees of all the inner transactions (which incorporates factors like higher fees for multisign or `AMMCreate`).
+
+The fees for the individual inner transactions are paid here instead of in the inner transaction itself, to ensure that fee escalation is calculated on the total cost of the transaction instead of just the overhead.
+
+### 2.2. `Flags`
+
+The `Flags` field represents the **batch mode** of the transaction. Exactly one must be specified in a `Batch` transaction.
+
+This spec supports four modes: 
+* `ALLORNOTHING` or `tfAllOrNothing` (with a value of `0x00000001`)
```

Comment:
You could optionally rename this to `ALL` (`tfAll`) which I think conveys the same idea as well or better. (The "nothing" case wouldn't really be a success in my book!)